### PR TITLE
created script for ebeam scan to multiprobe alignment

### PR DESCRIPTION
**Pull request by tepals**

### CODE REVIEW

File: scripts/ebeam_scan_to_multiprobe.py [hunk 1]
```diff
@@ -0,0 +1,92 @@
...
+    return:
+        phi (float): deflection angle in radians.
+    """
+    # Construct a system of linear equations to fit the line a*x + b*y + c = 0
+    n = coordinates.shape[0]
+    if n == 2:
+        x = coordinates[1, :] - coordinates[0, :]
+        # Determine the angle of this line to the x-axis, -pi/2 < phi < pi/2
+        phi = numpy.arctan2(x[0], x[1])
+    else:
```
[the remark this answers is not in the document]
Comment (tepals) [reply scripts/ebeam_scan_to_multiprobe.py]: The algorithm doesn't work when n==2.

File: scripts/ebeam_scan_to_multiprobe.py [hunk 2]
```diff
@@ -0,0 +1,92 @@
...
+
+    options = parser.parse_args(args[1:])
+    if options.debug:
+        logging.getLogger().setLevel(logging.DEBUG)
+    else:
+        logging.getLogger().setLevel(logging.WARNING)
+
+    ccd = ueye.Camera("camera", "ccd", device=None)
+    ccd.SetFrameRate(2)
+    image = getNextImage(ccd, timeout=5)
```
[the remark this answers is not in the document]
Comment (tepals) [reply scripts/ebeam_scan_to_multiprobe.py]: For now I've changed it to `image = ccd.data.get(asap=False)`, I will look into your last point independent of this PR

File: scripts/ebeam_scan_to_multiprobe.py [hunk 3]
```diff
@@ -0,0 +1,92 @@
...
+    if n == 2:
+        x = coordinates[1, :] - coordinates[0, :]
+        # Determine the angle of this line to the x-axis, -pi/2 < phi < pi/2
+        phi = numpy.arctan2(x[0], x[1])
+    else:
+        A = numpy.hstack((coordinates, numpy.ones((n, 1))))
+
+        # Solve the equation A*x = 0; i.e. find the null space of A. The solution is
+        # the eigenvector corresponding to the smallest singular value.
+        U, s, V = numpy.linalg.svd(A, full_matrices=False)
```
[the remark this answers is not in the document]
Comment (tepals) [reply scripts/ebeam_scan_to_multiprobe.py]: Andries will look into this

**Comment by tepals** [comment]
updated

**Comment by tepals** [comment]
Not ready to merge. Just tested the update and does not output the correct result, will check what's wrong.

File: scripts/ebeam_scan_to_multiprobe.py [hunk 4]
```diff
@@ -0,0 +1,94 @@
...
+
+    deflection_angle = get_deflection_angle(numpy.array(coordinates))
+    sem_rotation = deflection_angle - rotation
+    print("Apply {:.3f}° to the scan rotation using SEM PC".format(math.degrees(sem_rotation)))
+    return 0
+
+
+if __name__ == '__main__':
+    ret = main(sys.argv)
+    exit(ret)
```
Comment (tepals) [inline scripts/ebeam_scan_to_multiprobe.py]: @pieleric when I run the script through the terminal on Ubuntu 18.04 it does not exit, a keyboard interrupt also doesn't work. Do you know if I should use something else to exit?

**Comment by tepals** [comment]
Updated is working now.

File: scripts/ebeam_scan_to_multiprobe.py [hunk 5]
```diff
@@ -0,0 +1,117 @@
...
+
+def main(args):
+    """
+    Handles the command line arguments
+    args is the list of arguments passed
+    return (int): value to return to the OS as program exit code
+    """
+    parser = argparse.ArgumentParser()
+    parser.add_argument('--debug', action="store_true", default=False)
+    parser.add_argument("--role", dest="role", metavar="<component>", help="display and update an image on the screen")
```
Comment (pieleric) [inline scripts/ebeam_scan_to_multiprobe.py]: The `help` description is incorrect.

File: scripts/ebeam_scan_to_multiprobe.py [hunk 6]
```diff
@@ -0,0 +1,117 @@
...
+    if options.role:
+        if get_backend_status() != BACKEND_RUNNING:
+            raise ValueError("Backend is not running while role command is specified.")
+        ccd = model.getComponent(role=options.role)
+    else:
+        ccd = ueye.Camera("camera", "ccd", device=None)
+
+    try:
+        sem_rotation = get_sem_rotation(ccd)
+        print("Apply {:.3f}° to the scan rotation using SEM PC".format(math.degrees(sem_rotation)))
```
Comment (pieleric) [inline scripts/ebeam_scan_to_multiprobe.py]: I'd suggest to add an `except Execption`, which log the whole backtrace, and return 128.

**Comment by tepals** [comment]
updated

File: scripts/ebeam_scan_to_multiprobe.py [hunk 7]
```diff
@@ -0,0 +1,121 @@
...
+
+    deflection_angle = get_deflection_angle(numpy.array(coordinates))
+    sem_rotation = deflection_angle - rotation
+    return sem_rotation
+
+
+def main(args):
+    """
+    Handles the command line arguments
+    args is the list of arguments passed
```
Comment (Sabrina85) [inline scripts/ebeam_scan_to_multiprobe.py]: write same style as all other arguments/returns

File: scripts/ebeam_scan_to_multiprobe.py [hunk 8]
```diff
@@ -0,0 +1,121 @@
...
+            break
+
+    deflection_angle = get_deflection_angle(numpy.array(coordinates))
+    sem_rotation = deflection_angle - rotation
+    return sem_rotation
+
+
+def main(args):
+    """
+    Handles the command line arguments
```
Comment (Sabrina85) [inline scripts/ebeam_scan_to_multiprobe.py]: missing `"."`

**Pull request merged**